### ISSUE TITLE
`npm.ls()` v4.4.2 changed output, breaking package

### ISSUE THREAD

**Issue by dcporter** [wizardzloy/find-duplicate-dependencies#2]
It looks like the return value of `npm.ls()` changed substantially as of `npm` v4.4.2 — search for "Stop flattening ls-tree output" [here](https://github.com/npm/npm/releases/tag/v4.4.2). Same-version dependencies are no longer deduped in `npm.ls()` as they are on-disk, meaning this reports everything that's depended upon multiple times as a dup. Missing dependencies are also returned, with `missing: true` on them and `version: undefined`.

**Comment by kof** [comment]
I think I am seeing the same issue, I got printed out many times exact same version for e.g. 

```
  { name: 'warning',
    version: '3.0.0',
    from: 'warning@3.0.0',
    path: 'grape-web-client/grape-web/material-ui/react-event-listener' },
  { name: 'warning',
    version: '3.0.0',
    from: 'warning@3.0.0',
    path: 'grape-web-client/grape-web/material-ui/react-transition-group' },
  { name: 'warning',
    version: '3.0.0',
    from: 'warning@3.0.0',
    path: 'grape-web-client/grape-web/material-ui' },
```

**Comment by vovacodes** [comment]
I did not use this tool probably for 2 years already, so there are high chances it is not working with the current version of npm, unfortunately, I don't have much time to fix this now, but would appreciate and accept contributions

**Comment by kof** [comment]
what do you use instead?  maybe there is something else out there? Should be a popular problem …

**Comment by vovacodes** [comment]
Well I don't use anything at the moment, just rely on NPM and Webpack to dedupe as much as possible, but I agree that monitoring that stuff is quite useful

**Comment by kof** [comment]
The problem is how do we know that we need to update some dependency in order to avoid loading different versions if a dependency is used across multiple packages? Tools have no choice but  to bundle that dependency multiple times. This can't be automated afaik.

**Comment by vovacodes** [comment]
well this is indeed a very nice scenario of optimization, you almost convinced me to fix this bug and start using this script again :)

**Comment by kof** [comment]
I have a fix, give me a sec

**Comment by vovacodes** [comment]
fixed in `find-duplicate-dependencies@2.0.1`
